### PR TITLE
MAGECLOUD-4925: Ability to support custom database parameters in docker image

### PR DESCRIPTION
**Pull request by arhiopterecs**
### Fixed Issues (if relevant)

1. https://magento2.atlassian.net/browse/MAGECLOUD-4925

### Manual testing scenarios

1. Clone template: `git clone git@github.com:magento/magento-cloud.git`
2. Go to the project root directory: `cd magento-cloud`
3. Update dependencies: `composer update`
4. Build compose: `php vendor/bin/ece-docker build:compose`
5. Run containers: `docker-compose up -d`
6. Connect to `db` container: `docker-compose exec db bash`
7. Run cli command: `mysql -uroot -pmagento2 -e 'show variables WHERE Variable_name IN  ("collation_server", "character_set_server");' `
Expected result:
```
+----------------------+------------------+
| Variable_name        | Value            |
+----------------------+------------------+
| character_set_server | latin1           |
| collation_server     | latin1_swedish_ci|
+----------------------+------------------+
```
8. Create the file `./.docker/mysql/mariadb.conf.d/<any_name>.cnf` with next content:
```
# Use UTF8
[mysqld]
collation_server = utf8_general_ci
character_set_server = utf8
```
9. Run `docker-compose down -v && docker-compose up -d`
10. Connect to `db` container: `docker-compose exec db bash`
11. Run cli command: `mysql -uroot -pmagento2 -e 'show variables WHERE Variable_name IN  ("collation_server", "character_set_server");' `
Expected result:
```
+----------------------+-----------------+
| Variable_name        | Value           |
+----------------------+-----------------+
| character_set_server | utf8            |
| collation_server     | utf8_general_ci |
+----------------------+-----------------+
```
### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages

### CODE REVIEW

**Comment by aepod** [comment]
I would suggest putting a stub file with a comment at least in `.docker/mysql/mariadb.conf.d/example.cnf`

Not having this here will mean docker decides the permissions, which may leave it as root permissions. You need to have the directory at very least and I think the example.cnf will help newer people get started with modifying this.

Other than that this PR looks good.

**Comment by arhiopterecs** [comment]
> I would suggest putting a stub file with a comment at least in `.docker/mysql/mariadb.conf.d/example.cnf`
> 
> Not having this here will mean docker decides the permissions, which may leave it as root permissions. You need to have the directory at very least and I think the example.cnf will help newer people get started with modifying this.
> 
> Other than that this PR looks good.

@aepod 
Thank you for your review and for recommendations.
I added the example file with the comments

**Review by bbatsche — COMMENTED** [review]
What is the intended purpose of the `.gitignore` file? With no contents and the `example.cnf` file next to it, I'm not sure what it's doing.

**Comment by arhiopterecs** [comment]
Yes,  the `.gitignore` file can be removed

**Comment by arhiopterecs** [comment]
@bbatsche 
Yes, this file can be deleted

**Review by bbatsche — DISMISSED** [review]

File: dist/mysql/mariadb.conf.d/example.cnf [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+[client]
+# Default is Latin1, if you need UTF-8 set this (also in server section)
+#default-character-set = utf8
+#
+
+[mysqld]
+#
+# * Character sets
+#
```
Comment (NadiyaS) [inline dist/mysql/mariadb.conf.d/example.cnf:16]: Do we need these empty lines in comments? (for me it seems unnecessary - Lines: 8, 11, 13, 15)

File: dist/mysql/mariadb.conf.d/example.cnf [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+# Default is Latin1, if you need UTF-8 set all this (also in client section)
+#
+#character_set_server   = utf8
+#collation_server       = utf8_general_ci
```
Comment (NadiyaS) [inline dist/mysql/mariadb.conf.d/example.cnf]: Add empty line

File: dist/mysql/mariadb.conf.d/example.cnf (hunk 1 above)
Comment (arhiopterecs) [reply dist/mysql/mariadb.conf.d/example.cnf:16]: @NadiyaS Yes they should be

File: dist/mysql/mariadb.conf.d/example.cnf (hunk 2 above)
Comment (arhiopterecs) [reply dist/mysql/mariadb.conf.d/example.cnf]: @NadiyaS  Done

**Review by NadiyaS — DISMISSED** [review]

**Review by bbatsche — DISMISSED** [review]

**Comment by mveeramneni** [comment]
QA Approved

**Comment by mveeramneni** [comment]
QA Approved

**Review by shiftedreality — APPROVED** [review]

**Review by BaDos — APPROVED** [review]

**Pull request merged**